### DISCUSSION TITLE
"Us"/"them" terminology

### DISCUSSION THREAD

**Discussion by xendk** [magit/magit#4419] [Support] [answered]
I've noticed that I always get it wrong when magit talks about "us"/"them" or "ours"/"theirs". Is there a logic I fail to grasp?

For instance, I've just applied a stash, and it says "deleted by us", but I now I deleted them on the branch I'm applying the stash to. In this situation I would expect the stash to be "me" and the message to be "deleted by them".

Are "I" just the current branch whether you're merging, rebasing or appyling stashes, or is there some other logic?

**Comment by tarsius** [comment, marked as answer]
It's git terminology. Some of the answers on https://stackoverflow.com/questions/25576415/what-is-the-precise-meaning-of-ours-and-theirs-in-git might help.

**Reply by xendk** [reply]
Ah, thanks, that does clear things up. So "ours" is what we're merging into (and popping a stash is basically merging into the current branch workspace). Except for rebase where "ours" is what you're rebasing onto. Kinda makes sense when explained, now I just got to remember it.

**Reply by tarsius** [reply]
Ours is what we already have. Theirs is the incoming stuff. Before rebasing we have to check out "their" branch, which is confusing because we would normally start calling the checked out branch "our" side after doing that. However before "their" commits are being applied, git has to internally "rewind" to "our" side before applying "their" commits on top of that; so right before commits are being applied "our" side is checked out (as always). It's just that in order to rebase we have to check out their side, which is the part that makes things confusing. Well that and the fact that the commits being rebases are usually on a feature branch that "we" have authored.

**Reply by felipeochoa** [reply]
This reads like who's on first :smile:

The website linked from one of the answers (https://nitaym.github.io/ourstheirs/) was helpful for me

Do you know if git is consistent in having ours/theirs as upper/lower? If so, might be worth adding that to the `read-char-case` menu. Like `(?o "[o]ur stage (upper)"   "--ours")`

**Reply by xendk** [reply]
But then you'd have to remeber whether the stash is upper or lower. I think the best solution would be naming things properly, i.e. `main`, `my-feature-branch`, `stash`), but that probably requires changes in many places. Not to mention the issue of what to call things. When cherry-picking, do you call it 'cherry' or use the commit hash? Or worse, you're merging a random commit into a detached head...

**Reply by felipeochoa** [reply]
With upper/lower at least the per-hunk reverting is easier, since you can see which is which

**Reply by tarsius** [reply]
Naturally.

**Reply by xendk** [reply]
> With upper/lower at least the per-hunk reverting is easier, since you can see which is which

But if you revert a deleted file, magit ask which one to revert to, but there's no upper or lower as there is no hunk.

Naming things by branch/sha/etc removes one level of mental redirect, but I appreciate that it's more involved to implement.
